biomage/experiment/pull.py

```diff
@@ -72,26 +106,54 @@ def update_config_if_needed(filepath, table_name, experiment_id):
     # locally in the API. This solution is not ideal as it will fail
     # if the field name changes or more tightly coupled info is added
     # TODO: make api handle not found cases, or ignore keys in development env
-    if "experiment" in table_name:
-        remote_cfg = remove_key(remote_cfg, "pipeline")
+    remote_cfg = remove_key(remote_cfg, "pipeline")
+
+    # add the cognito user specified in the env to the experiment permissions
+    remote_cfg = add_env_user_to_experiment(cfg=remote_cfg)
 
     # if the local config was not found or it's different from the remote => update
     if not found or local_cfg != remote_cfg:
         save_cfg_file(remote_cfg, filepath)
         Summary.add_changed_file(filepath)
 
+    return remote_cfg["projectId"]["S"]
 
-def update_configs(experiment_id, origin):
-    # config pairs like: (local file name, remote table name)
-    configs = [
-        (EXPERIMENTS_FILE, EXPERIMENTS_TABLE),
-        (SAMPLES_FILE, SAMPLES_TABLE),
-    ]
 
-    for file_name, table_name in configs:
-        file_path = f"{experiment_id}/{file_name}"
-        table_name = f"{table_name}-{origin}"
-        update_config_if_needed(file_path, table_name, experiment_id)
+def update_config_if_needed(filepath, table_name, experiment_id):
+    """
+    Filepath: experiment_id/filename
+    """
+
+    local_cfg, found = load_cfg_file(filepath)
+
+    client = boto3.client("dynamodb")
+    remote_cfg = client.get_item(
+        TableName=table_name, Key={"experimentId": {"S": experiment_id}}
+    )["Item"]
+
+    # if the local config was not found or it's different from the remote => update
+    if not found or local_cfg != remote_cfg:
+        save_cfg_file(remote_cfg, filepath)
+        Summary.add_changed_file(filepath)
+
+
+def update_configs(experiment_id, origin):
+    # update samples
+    file_path = f"{experiment_id}/{SAMPLES_FILE}"
+    table_name = f"{SAMPLES_TABLE}-{origin}"
+    project_uuid = update_config_if_needed(file_path, table_name, experiment_id)
+
+    # update experiments
+    file_path = f"{experiment_id}/{EXPERIMENTS_FILE}"
+    table_name = f"{EXPERIMENTS_TABLE}-{origin}"
+    project_uuid = update_experiment_config_if_needed(
```

Comment:
Why are you getting `project_uuid` assigned to both on line 144 and on line 149? Makes me worried what would happen if in the future we swap those two lines -- will that make `project_uuid` have a wrong value?